pkg/module/file/file.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2025 Blindspot Software
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package file provides a dutagent module that transfers files between client and dutagent.
+package file
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/BlindspotSoftware/dutctl/pkg/module"
+)
+
+func init() {
+	module.Register(module.Record{
+		ID: "file",
+		New: func() module.Module {
+			return &File{}
+		},
+	})
+}
+
+// File permission constants.
+const (
+	defaultDirPerm  = 0o755 // Default directory permissions
+	defaultFilePerm = 0o644 // Default file permissions
+)
+
+// op represents the file operation.
+type op string
+
+const (
+	opUpload   op = "upload"
+	opDownload op = "download"
+)
+
+// File is a module that transfers files between client and dutagent.
+type File struct {
+	// Permission sets file permissions in octal format (e.g., "0644", "0755").
```

Comment:
Use Markdown-link to the default constant. (See how to use links in doc-comments.)